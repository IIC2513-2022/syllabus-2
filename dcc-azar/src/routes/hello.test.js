const supertest = require('supertest');
const app = require('../app');

const request = supertest(app.callback());

describe('GET /hello', () => {
  test('responds correctly', async () => {
    const response = await request.get('/hello');

    expect(response.status).toBe(200);
    expect(response.text).toMatchSnapshot();
  });
});

describe('POST /hello', () => {
  test('responds correctly', async () => {
    const response = await request.post('/hello');
    expect(response.status).toBe(302);
  });
});

describe('GET /hello/:name', () => {
  test('responds correctly', async () => {
    const response = await request.get('/hello/my-name');

    expect(response.status).toBe(200);
    expect(response.body).toEqual({ message: 'Hello my-name!' });
  });
});
