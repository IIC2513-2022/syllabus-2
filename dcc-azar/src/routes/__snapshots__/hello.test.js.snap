// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /hello responds correctly 1`] = `
"<!DOCTYPE html>
<html>

  <head>
    <meta charset=\\"utf-8\\">
    <title>IIC2513 - CHANGE-ME</title>
    
      <link rel=\\"stylesheet\\" href=\\"/assets/app.css\\">
    
  </head>

  <body>
    <header>
      <div id=\\"logo\\">
        <img src=\\"/assets/logo.png\\" alt=\\"Logo\\" />
      </div>
    </header>
    

<p>
  Helloâ€¦ <a href=\\"/hello/your-name-here\\">your-name-here</a>.
</p>

<p>
  You can also fill this form if you feel like doing so...
  <form action=\\"/hello\\" enctype=\\"multipart/form-data\\" method=\\"post\\">
    <input type=\\"text\\" name=\\"username\\">
    <input type=\\"password\\" name=\\"password\\">
    <input type=\\"file\\" name=\\"uploads\\" multiple=\\"multiple\\"><br>
    <input type=\\"submit\\" value=\\"Subir\\">
  </form>
</p>

    <script src=\\"/assets/app.js\\"></script>
  </body>

</html>
"
`;
